pkg/cmd/cli/secrets/known_secret_types.go

```diff
@@ -12,13 +12,11 @@ type KnownSecretType struct {
 	RequiredContents util.StringSet
 }
 
-func (ks KnownSecretType) Matches(secretContent map[string][]byte) bool {
+func (ks KnownSecretType) Matches(secretContent []string) bool {
 	if secretContent == nil {
 		return false
 	}
-
-	secretKeys := util.KeySet(reflect.ValueOf(secretContent))
-	return reflect.DeepEqual(ks.RequiredContents.List(), secretKeys.List())
+	return reflect.DeepEqual(ks.RequiredContents.List(), secretContent)
```

Comment:
@deads2k your comment got lost in the rebase. The reason this is for `[]string` is that when we allow users to give their own names, we can no longer glean information from their file names from the `map[string][]byte` because it no longer contains any references to their file names. Should there be a parallel "overloaded" `MatchesMap(data []byte)` function?